src/main/java/io/neonbee/data/DataVerticle.java

```diff
@@ -343,10 +343,14 @@ public static <U> Future<U> requestData(Vertx vertx, DataRequest request, DataCo
                         LOGGER.correlateWith(context).debug("Received event bus reply");
 
                         if (asyncReply.succeeded()) {
+                            DataContext responseDataContext =
+                                    decodeContextFromString(asyncReply.result().headers().get(CONTEXT_HEADER));
                             context.setData(Optional
                                     .ofNullable(
                                             decodeContextFromString(asyncReply.result().headers().get(CONTEXT_HEADER)))
                                     .map(DataContext::data).orElse(null));
```

Comment:
decodeFromString should not be called twice here, because it is kind of a heavy operation, but maybe it is not needed at all, see comment below